packages/core/src/App/Components/Layout/Header/toggle-menu-drawer.jsx

```diff
@@ -14,163 +13,95 @@ import { observer, useStore } from '@deriv/stores';
 import { localize, getAllowedLanguages, Localize } from '@deriv/translations';
 import NetworkStatus from 'App/Components/Layout/Footer';
 import ServerTime from 'App/Containers/server-time.jsx';
-import { BinaryLink, LanguageLink } from 'App/Components/Routes';
+import { LanguageLink } from 'App/Components/Routes';
 import getRoutesConfig from 'App/Constants/routes-config';
-import { changeLanguage } from 'Utils/Language';
 import LiveChat from 'App/Components/Elements/LiveChat';
 import useLiveChat from 'App/Components/Elements/LiveChat/use-livechat.ts';
 import PlatformSwitcher from './platform-switcher';
+import { MenuLink } from './menu-link';
+
+const MobileLanguageMenu = React.memo(
```

Comment:
@yauheni-deriv `React.memo` is not needed once you use `useStore` here and wrap it with `observer`.